### PR TITLE
Add mutable mapping scrub function parameter 

### PR DESCRIPTION
**Pull request by matheusbrat**

### CODE REVIEW

**Comment by untitaker** [comment]
Sorry I should've been clearer. I think I want this interface:

```
def callback(d):
    del d['foo']

@sensitive_variables(variable_name=callback)
def foo():
    variable_name = {"foo": 42}
```

I don't really want to special-case mutablemapping too much (this can also become an issue with arrays), ideally we would allow users to just define custom scrubbing per var name with this.

**Comment by untitaker** [comment]
Ah I see, kwargs are already used (I misremembered this). What about changing this patch to always call the callback for all types, and also providing the variable name as argument?

**Comment by matheusbrat** [comment]
Just to be sure I understood it right. You want to be able to define a "custom_scrub_fn" which is always called and one of the arguments is the variable name (like the local being processed).

With this approach we need to think on types passed as reference such as dict and other types, I believe we would need to define the fn to return "bool, new_value" so it could update anything.

Maybe something like:
```
            for local in locals:
                obj = locals[local]
                locals_modified, locals[local] = custom_scrub_fn(local, obj)
```

What do you think?

**Comment by untitaker** [comment]
Yes that makes sense! Please also update the README

**Comment by matheusbrat** [comment]
@untitaker, what do you think from the current implementation?

File: README.md [hunk 1]
```diff
@@ -30,9 +34,40 @@ results in:
...
+    print("Logging in " + username + " with " + password)
+
+# TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
+login_user(None, "secret123")
+```
+
 ## How does it work?
 
 When the decorated function throws an exception, `sensitive_variables` walks through the traceback, removes sensitive data from `frame.f_locals` and reraises the exception.
+If mutablemapping_scrub_fn is set and there is a MutableMapping instance on locals, it will call the function with the instance as parameter.
```
Comment (untitaker) [inline README.md]: pls fix the function name reference here, otherwise lgtm

**Review by untitaker — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (matheusbrat) [reply README.md]: I adjusted the text a bit. Please take a look.

**Comment by untitaker** [comment]
Thanks!

**Pull request merged**